### PR TITLE
[380] enhance SonataFlowClusterPlatform.spec

### PR DESCRIPTION
**Pull request by tchughesiv**
**Description of the change:**
Enhance SonataFlowClusterPlatform.spec to better communicate which capabilities are used by workflows cluster-wide. Adds `capabilities.workflow["services"]` to `spec`.

```yaml
apiVersion: sonataflow.org/v1alpha08
kind: SonataFlowClusterPlatform
metadata:
  name: cluster-platform
spec:
  platformRef:
    name: sonataflow-platform
    namespace: central-platform-ns
  capabilities:
    workflow:
    - services
```

**Motivation for the change:**
fixes https://github.com/apache/incubator-kie-kogito-serverless-operator/issues/380

**Checklist**

- [ ] Add or Modify a unit test for your change
- [ ] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Review by ricardozanini — APPROVED** [review]

**Review by wmedvede — APPROVED** [review]
LGTM.

But, unless we are sure that other "capabilities" will managed at cluster level in the short term, and we have them more or less identified, I don't see why we need this. 
I mean, right now, if you defile a SFCP is because you want to make DI and JS cluster wide, so, if you don't won't this capability, you don't define a SFCP, that's it.

**Comment by ricardozanini** [comment]
@wmedvede we can't preview the future atm, @wmedvede. But the core engine is already based on those capabilities in the form of Quarkus extensions. So, for instance, I can see cluster-level messaging, persistence, monitoring, and so on in the future. So leaving the API capable of accommodating those is the intention of this PR.

**Comment by tchughesiv** [comment]
> @tchughesiv any plans to add E2E tests to this PR? So far there are none for this controller.

@jordigilh @ricardozanini should i add them to this PR or a separate one?

**Comment by jordigilh** [comment]
> > @tchughesiv any plans to add E2E tests to this PR? So far there are none for this controller.
> 
> @jordigilh @ricardozanini should i add them to this PR or a separate one?

Any reason for not adding them to this PR already? Perhaps dependencies to other unmerged PRs?

**Review by jakubschwan — APPROVED** [review]
lgtm

**Comment by tchughesiv** [comment]
> Any reason for not adding them to this PR already? Perhaps dependencies to other unmerged PRs?

@jordigilh let me see what i can do

**Comment by tchughesiv** [comment]
@jordigilh @ricardozanini i've added e2e tests

File: test/e2e/clusterplatform_test.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+				cmd = exec.Command("kubectl", "get", "sfplatform", "-n", targetNamespace2, "sonataflow-platform", "-o", "jsonpath='{.status.clusterPlatformRef.services.jobServiceRef.url}'")
+				returnedValue, _ := utils.Run(cmd)
+				return returnedValue
+			}, 20*time.Minute, 5).Should(Equal([]byte("'" + jobServiceUrl + "'")))
+
+			cmd = exec.Command("kubectl", "delete", "SonataFlowClusterPlatform", "cluster", "--wait")
+			_, err = utils.Run(cmd)
+			Expect(err).NotTo(HaveOccurred())
+		},
+			Entry("without services configured", test.GetSonataFlowE2EPlatformNoServicesDirectory(), dev, ephemeral),
```
Comment (jordigilh) [inline test/e2e/clusterplatform_test.go:155]: Do you think it's worth covering additional use cases, for instance when one of the services is already provided in the target namespace, or when both are?

Comment (tchughesiv) [reply test/e2e/clusterplatform_test.go]: done

**Review by ricardozanini — APPROVED** [review]

**Comment by ricardozanini** [comment]
@tchughesiv can you please rebase?

**Comment by tchughesiv** [comment]
@ricardozanini done
